src/cfnlint/runner.py

```diff
@@ -434,8 +434,14 @@ def cli(self) -> None:
             sys.exit(0)
 
         if not self.config.templates:
+            self.config.parser.print_help()
+
+            # If templates were specified, but didn't create any file matches, print out message and exit
+            template = self.config.get_input_templates
+            if template:
+                print(f"\nNo matching files found for template arguments: {template}")
+                return
             if sys.stdin.isatty():
```

Comment:
Doesn't this need to be moved up to support piping a template to cfn-lint?

Reply:
OK I understand a bit more about the project now.

The `templates` argument can either be piped in via STDIN or a string containing a list of cfn template files.

I'll push up an update shortly.

Reply:
After a few updates to the `templates` property of `ConfigMixIn`, I worked out that the property doesn't get passed in `stdin` and instead `stdin` is processed by `fileinput` in the `cfn_yaml` or `cfn_json` modules.

Hence we don't need to worry about 'stdin' within the context of 'templates'